--- Makefile	2016-06-29 09:40:41.411673300 -0400
+++ ../Makefile	2016-06-29 09:38:47.233256600 -0400
@@ -18,18 +18,33 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 # IN THE SOFTWARE.

+VERSION = 2.7.0
 PLATFORM ?= $(shell sh -c 'uname -s | tr "[A-Z]" "[a-z]"')
+ifeq ($(findstring cygwin,$(PLATFORM)), cygwin)
+PLATFORM = winnt
+endif
+ifeq ($(findstring mingw,$(PLATFORM)), mingw)
+PLATFORM = winnt
+endif
+ifeq ($(findstring msys,$(PLATFORM)), msys)
+PLATFORM = winnt
+endif
+
 HELPER ?=
 BINEXT ?=
 ifeq (darwin,$(PLATFORM))
-SONAME ?= libhttp_parser.2.7.0.dylib
+SONAME ?= libhttp_parser.$(VERSION).dylib
 SOEXT ?= dylib
 else ifeq (wine,$(PLATFORM))
 CC = winegcc
 BINEXT = .exe.so
 HELPER = wine
+else ifeq (winnt,$(PLATFORM))
+CC = gcc
+SONAME ?= libhttp_parser.$(VERSION).dll
+SOEXT  ?= dll
 else
-SONAME ?= libhttp_parser.so.2.7.0
+SONAME ?= libhttp_parser.so.$(VERSION)
 SOEXT ?= so
 endif

@@ -50,7 +65,12 @@
 CFLAGS_DEBUG = $(CFLAGS) -O0 -g $(CFLAGS_DEBUG_EXTRA)
 CFLAGS_FAST = $(CFLAGS) -O3 $(CFLAGS_FAST_EXTRA)
 CFLAGS_BENCH = $(CFLAGS_FAST) -Wno-unused-parameter
+
+ifneq (winnt,$(PLATFORM))
+CFLAGS_LIB = $(CFLAGS_FAST)
+else
 CFLAGS_LIB = $(CFLAGS_FAST) -fPIC
+endif

 LDFLAGS_LIB = $(LDFLAGS) -shared

@@ -141,7 +161,7 @@
 	rm -f *.o *.a tags test test_fast test_g \
 		http_parser.tar libhttp_parser.so.* \
 		url_parser url_parser_g parsertrace parsertrace_g \
-		*.exe *.exe.so
+		*.exe *.exe.so *.dll

 contrib/url_parser.c:	http_parser.h
 contrib/parsertrace.c:	http_parser.h
